src/Xamarin.Android.Build.Tasks/Tasks/GetAndroidPackageName.cs

```diff
@@ -49,18 +49,20 @@ public class GetAndroidPackageName : AndroidTask
 
 		public override bool RunTask ()
 		{
-			if (!string.IsNullOrEmpty (PackageName)) {
-				PackageName = AndroidAppManifest.CanonicalizePackageName (PackageName);
-			} else if (!string.IsNullOrEmpty (ManifestFile) && File.Exists (ManifestFile)) {
+			if (!string.IsNullOrEmpty (ManifestFile) && File.Exists (ManifestFile)) {
 				using var stream = File.OpenRead (ManifestFile);
 				using var reader = XmlReader.Create (stream);
 				if (reader.MoveToContent () == XmlNodeType.Element) {
 					var package = reader.GetAttribute ("package");
 					if (!string.IsNullOrEmpty (package)) {
 						package = ManifestDocument.ReplacePlaceholders (ManifestPlaceholders, package);
 						PackageName = AndroidAppManifest.CanonicalizePackageName (package);
+					} else if (!string.IsNullOrEmpty (PackageName)) {
```

Comment:
Instead of this `else if` here, and the "duplicate" `PackageName = …` statement between the two blocks, why not *remove* this `else if`, and turn the `else if` on line 64 into an `if`?

Reply:
I moved it around, should be better now. I removed a `CanonicalizePackageName()` call as well, so it isn't called twice.